### PR TITLE
Feature/4 dof for point to plane minimizer

### PR DESCRIPTION
**Pull request by kubelvla**
This PR adds the possibility to optimize only translation and yaw angle instead of the full 6DOF pose. A unit test has been also added. Since this feature is a part of research in progress, there are no publications yet. The code has been successfully used during the DARPA SubT challenge, so I consider it quite well tested.

### CODE REVIEW

**Comment by pomerlef** [comment]
add to whitelist

File: examples/data/icp_data/force4DOFForPointToPlaneMinimizer.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+referenceDataPointsFilters:
+  - SurfaceNormalDataPointsFilter:
+      knn: 15
+
+  - ObservationDirectionDataPointsFilter
+
+  - OrientNormalsDataPointsFilter:
+      towardCenter: 1
+
```
Comment (YoshuaNava) [inline examples/data/icp_data/force4DOFForPointToPlaneMinimizer.yaml:16]: Are the extra lines between the reference data points filters intended?

**Comment by YoshuaNava** [comment]
I just approved this PR, it's a really cool contribution.

Just out of curiosity: did you observe any improvement in the CPU usage or accuracy of ICP with respect to full DOF minimization?

File: examples/data/icp_data/force4DOFForPointToPlaneMinimizer.yaml (hunk 1 above)
Comment (kubelvla) [reply examples/data/icp_data/force4DOFForPointToPlaneMinimizer.yaml:16]: I like it better, easier to distinguish between the filters.

**Comment by kubelvla** [comment]
There should not be a significant speedup, the difference is solving a 4x4 matrix instead of 6x6 one. I don't expect much speedup, the heavy lifting is done elsewhere in ICP...

**Comment by YoshuaNava** [comment]
Ok. All good.

**Comment by YoshuaNava** [comment]
@pomerlef Can this PR be merged?

**Comment by pomerlef** [comment]
All good, go ahead

**Comment by kubelvla** [comment]
@simonpierredeschenes Simon Pierre, could you press the button please? Thx

**Pull request merged**